app/src/main/java/com/alexstyl/themingandroid/AppCompatThemedActivity.java

```diff
@@ -38,10 +38,10 @@ protected final void applyTheme(AppTheme theme) {
     }
 
     private void restartActivity() {
-        Intent intent = getIntent();
+        final Intent intent = getIntent();
         intent.removeCategory(Intent.CATEGORY_LAUNCHER);
-        startActivity(intent);
         finish();
+        startActivity(intent);
```

Comment:
This came up by trial and error really. I noticed that the previous order wouldn't work when called from inside a fragment. But this order works in every situation.